geode-core/src/main/java/org/apache/geode/internal/cache/tier/CommunicationMode.java

```diff
@@ -22,6 +22,19 @@
  * query methods must be updated.
  */
 public enum CommunicationMode {
+  /**
+   * The first byte of any locator connection will be the high order byte of its gossip version,
+   * which will always be 0. Communication modes should not collide with this value.
+   */
+  ReservedForGossip((byte) 0, "Locator gossip version"),
+  /**
+   * For the new client-server protocol.
+   *
+   * Protobuf handshake messages are specially constructed so that this value will match the first
+   * byte sent, allowing clients to start protobuf connections with a handshake instead of
+   * communication mode bytes.
+   */
+  ProtobufClientServerProtocol((byte) 10, "Protobuf client"),
```

Comment:
Will this require coordination with the C# and Go drivers?

Reply:
Kind of, they'll need to back out any magic byte they're currently sending us and just start each connection with a Handshake message.